shared/cluster.go

```diff
@@ -409,3 +409,139 @@ func WriteDataPod(namespace string) (string, error) {
 
 	return RunCommandHost(cmd)
 }
+
+type ServiceType string
+
+const (
+	Server ServiceType = "server"
+	Agent  ServiceType = "agent"
+)
+
+func (s ServiceType) IsValidServiceType() bool {
+	valid := []ServiceType{Server, Agent}
+	for _, i := range valid {
+		if s == i {
+			return true
+		}
+	}
+
+	return false
+}
+
+type ServiceAction string
+
+const (
+	Stop    ServiceAction = "stop"
+	Start   ServiceAction = "start"
+	Restart ServiceAction = "restart"
+	Status  ServiceAction = "status"
+)
+
+func (s ServiceAction) IsValidServiceAction() bool {
+	valid := []ServiceAction{Stop, Start, Restart, Status}
+	for _, i := range valid {
+		if s == i {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (s ServiceAction) isStartAction() bool {
+	startAction := []ServiceAction{Start, Restart}
+	for _, i := range startAction {
+		if s == i {
+			return true
+		}
+	}
+
+	return false
+}
+
+type Product string
+
+const (
+	K3S  Product = "k3s"
```

Comment:
We should already have the product from .env file, isnt it?

Reply:
just trying to create an enum kind of class - and have all methods related to k3s/rke2 here. 
we do get the product from .env => based on which we set the product := shared.K3S or product := shared.RKE2

and we run commands like 
product.ManageClusterService -> for start/stop k3s or rke2 etc. 

similarly product.CertRotate

since all these are really commands we are running on our k3s/rke2 service/product.

Reply:
will talk about this tomm in the standup.

Reply:
https://github.com/rancher/distros-test-framework/pull/42/files#r1397327139

Reply:
> just trying to create an enum - ensuring unique values and also to have all methods related to k3s/rke2 here.  

also this validation is already done on the runtime   https://github.com/rancher/distros-test-framework/blob/dbc0224ca28dba1bd9567c33a0a89112ed2904bf/config/reader.go#L33C2-L54C1

Reply:
Yeah if we want to create Enums, then lets do it at the start of the cluster creation cause during test run time, it wont change